### ISSUE TITLE
Cannot find module 'unplugin-vue-macros/vite' or its corresponding type declarations

### ISSUE THREAD

**Issue by mxp131011** [sxzz/unplugin-vue-macros#257]
### Describe the bug

![image](https://user-images.githubusercontent.com/31055355/215719979-f826fb24-308c-4dc8-91aa-3c842ab922bc.png)

##  tsconfig.json
```js
{
  "compilerOptions": {
    "baseUrl": "./",
    // "outDir": "./dist", // 指定输出目录
    "target": "ESNext", // 目标语言的版本
    "module": "ESNext", // 生成代码的模板标准
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "isolatedModules": false,
    "preserveValueImports": true,
    "allowSyntheticDefaultImports": true,
    "jsx": "preserve",
    "declaration": false, // 生成声明文件，开启后会自动生成声明文件
    "types": ["unplugin-vue-macros/macros-global", "vite/client", "element-plus/global", "vite-plugin-pwa/client"],
    "useDefineForClassFields": true,
    "lib": ["ESNext", "DOM", "DOM.Iterable"], // TS需要引用的库，即声明文件，es5 默认引用dom、es5、scripthost,如需要使用es的高级版本特性，通常都需要配置，如es8的数组新特性需要引入"ES2019.Array",
    "strict": true, // 启用所有严格类型检查选项
    "noImplicitAny": true, // 在表达式和声明上有隐含的 any类型时报错
    "strictNullChecks": true, // 当类型检查时,要考虑' null '和' undefined '。
    "noImplicitThis": true, // 当 this 表达式值为 any 类型的时候,生成一个错误
    "alwaysStrict": true, // 以严格模式检查每个模块,并在每个文件里加入 'use strict'
    "resolveJsonModule": true, // 启用.json文件导入
    "allowJs": true, // 允许编译器编译JS，JSX文件
    "checkJs": true, // 允许在JS文件中报错，通常与allowJS一起使用
    "noUnusedLocals": true, // 有未使用的变量时,抛出错误
    "noUnusedParameters": true, // 有未使用的参数时,抛出错误
    "noImplicitReturns": true, // 并不是所有函数里的代码都有返回值时,抛出错误
    "noFallthroughCasesInSwitch": true, // 报告 switch 语句的 fallthrough 错误。(即,不允许 switch 的 case 语句贯穿)
    "useUnknownInCatchVariables": false, // 类型捕获子句变量为'unknown'而不是'any'
    "removeComments": true, // 移除注释
    "noEmit": true, // 禁止编译器生成文件
    "newLine": "lf", // 设置生成文件的换行符
    "forceConsistentCasingInFileNames": true, // 强制文件名的大小写一致
    "strictFunctionTypes": true, // 在赋值函数时,请检查以确保参数和返回值是子类型兼容的
    "strictBindCallApply": true, // 检查' bind '、' call '和' apply '方法的参数是否与原始函数匹配
    "strictPropertyInitialization": true, // 检查在构造函数中声明但没有设置的类属性
    "noUncheckedIndexedAccess": true, // 在索引签名结果中包含'undefined'
    "noImplicitOverride": true, // 确保派生类中的重写成员用重写修饰符标记
    "importsNotUsedAsValues": "error", // 强制对类型使用“import type”而不是“import”
    "paths": {
      "@/*": ["src/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.d.ts", "src/**/*.tsx", "src/**/*.vue", "vite.config.ts", ".eslintrc.js"]
}

```
##  package.json
```js
{
  "name": "zlt9_erp_pc",
   
  "license": "private",
  "private": true,
  "scripts": {
     
    "tsc": "vue-tsc --noEmit --skipLibCheck",
    "eslint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
    "stylelint": "stylelint src/**/*.{css,scss,vue} --fix"
  },
  "dependencies": {
    "@formkit/auto-animate": "1.0.0-beta.6",
    "@vueuse/core": "^9.12.0",
    "accessible-nprogress": "^2.1.2",
    "axios": "^1.2.6",
    "crypto-js": "^4.1.1",
    "echarts": "^5.4.1",
    "element-plus": "^2.2.28",
    "jsbarcode": "^3.11.5",
    "mitt": "^3.0.0",
    "pinia": "^2.0.29",
    "pinia-plugin-persistedstate": "^3.0.2",
    "qrcode": "^1.5.1",
    "super-localstorage": "^2.0.9",
    "vue": "^3.2.45",
    "vue-router": "^4.1.6",
    "vuedraggable": "^4.1.0",
    "xlsx": "^0.18.5"
  },
  "devDependencies": {
    "@types/crypto-js": "^4.1.1",
    "@types/qrcode": "^1.5.0",
    "@vitejs/plugin-vue": "^4.0.0",
    "@vue/compiler-sfc": "^3.2.45",
    "@vue/eslint-config-prettier": "^7.0.0",
    "@vue/eslint-config-typescript": "^11.0.2",
    "eslint": "^8.33.0",
    "eslint-plugin-jsdoc": "^39.7.4",
    "eslint-plugin-vue": "^9.9.0",
    "postcss": "^8.4.21",
    "postcss-html": "^1.5.0",
    "prettier": "^2.8.3",
    "sass": "^1.57.1",
    "stylelint": "^14.16.1",
    "stylelint-config-html": "^1.1.0",
    "stylelint-config-standard-scss": "^6.1.0",
    "stylelint-order": "^6.0.1",
    "typescript": "^4.9.5",
    "unplugin-auto-import": "^0.13.0",
    "unplugin-vue-components": "^0.23.0",
    "unplugin-vue-macros": "^1.7.2",
    "vite": "^4.0.4",
    "vite-auto-deploy": "^1.1.19",
    "vite-plugin-banner": "^0.7.0",
    "vite-plugin-pwa": "^0.14.1",
    "vue-tsc": "^1.0.24"
  },
  "publishConfig": {
    "registry": "https://registry.npmmirror.com/"
  }
}

```

### System Info

```Shell
System:
    OS: Windows 10 10.0.22621
    CPU: (4) x64 Intel(R) Core(TM) i3-8100 CPU @ 3.60GHz
    Memory: 7.46 GB / 15.94 GB
  Binaries:
    Node: 18.13.0 - D:\IDE\nodejs\node.EXE
    npm: 8.19.3 - D:\IDE\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1105.0), Chromium (109.0.1518.70)
    Internet Explorer: 11.0.22621.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sxzz** [comment]
`moduleResolution` 改为 `node16`
Change `moduleResolution` to `node16`

**Comment by mxp131011** [comment]
> `moduleResolution` 改为 `node16` Change `moduleResolution` to `node16`

这会影响其他模块如：vite 和 unplugin-auto-import/vite ，我记得1.6.4是没问题的

**Comment by sxzz** [comment]
方案1. 暂时使用 `// @ts-expect-error` 忽略错误
方案2. 升级到 TypeScript 5.0 Beta，并将 `moduleResolution` 修改为 `bundler`
方案3. 降级到 1.6.4 版本，等待 TypeScript 发布 5.0 正式版后尝试方案 2

Option A. temporarily use `// @ts-expect-error` to ignore errors
Option B. Upgrade to TypeScript 5.0 Beta and change `moduleResolution` to `bundler`.
Option C. Downgrade to version 1.6.4 and try option B after TypeScript 5.0 is released

**Comment by sxzz** [comment]
`unplugin-auto-import/vite` 应该是没有问题的，如果有问题可以发issue（附带 minimum repo） 到 unplugin-auto-import 的仓库。

**Comment by so1ve** [comment]
还有一个办法，就是直接导入整个Unplugin：
```ts
import VueMacros from "unplugin-vue-macros";

export default defineConfig({
  plugins: [
    VueMacros.vite(/* ... */),
  ],
})
```

**Comment by YAN7** [comment]
引入路径改成这样也行
```ts
import DefineOptions from "unplugin-vue-define-options/dist/vite";
```

**Comment by wangliguokaka** [comment]
tsconig.json文件中 把"moduleResolution":设置为"nodenext"


{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "nodenext",
    "strict": false,
    "jsx": "preserve",
    "importHelpers": true,
 .........

**Comment by sxzz** [comment]
Fixed in https://github.com/sxzz/unplugin-vue-macros/commit/e33953564dbbce28d4b61f61b27d7cdfcdf5c242

**Comment by xiaozhangtongx** [comment]
方案2. 升级到 TypeScript 5.0 Beta，并将 moduleResolution 修改为 bundler，
方案1可行，但使用方案2时，`"typescript": "^5.0.2"`，`element-plus/global`会报错，请问大佬这该怎么解决
![image](https://user-images.githubusercontent.com/63902256/226097030-ddf1a1b4-1cad-4334-a95c-5a96528c08a0.png)
具体版本如下：
```json
    "unplugin-vue-define-options": "^1.2.4",
    "vite": "^4.1.4",
    "vitest": "^0.29.3",
    "vue-tsc": "^1.2.0"
    "typescript": "^5.0.2",
    "element-plus": "^2.3.0",
```

**Comment by sxzz** [comment]
此为 Element Plus 的 bug，请到该仓库反馈。

**Comment by xiaozhangtongx** [comment]
> 此为 Element Plus 的 bug，请到该仓库反馈。
好的谢谢大佬，马上就去！

**Comment by pany-ang** [comment]
1. 如果 ts 版本小于 5，那么出现这个告警的原因应该是最新版 vscode，测试后发现好像 vscode 1.74 以及更早的版本不会出现该告警（这种情况下，你几乎只能忍着这个警告不管，其他措施会导致 build 报错，或者降低你的 vscode 版本）

2. 如果 ts 版本大于 5，那么你可以选择将 moduleResolution 配置为 bundler，但是这样的话，其他依赖（比如 Element Plus）可能就会疯狂报错（等待官方修复这个问题中...）。你也可以选择“// @ts-expect-error”来忽略这个错误，忽略后能正常 build

3. ts 小于 5，并且又是最新版 vscode 时最难受，几乎处理不了这个报错

所以总结一下，如果 ts 小于 5，那么就不要去管它，issues 里面的别人给的方法，几乎都会导致 build 出错。如果 ts 大于 5，那么就暂时用“// @ts-expect-error”来忽略这个错误，这样能正常 build

1. If the TS version is less than 5, then the reason for this alarm should be the latest version of VSCODE, after testing, it seems that VSCODE 1.74 and earlier versions will not appear this alarm (in this case, you can almost only endure this warning, other measures will cause build errors, or reduce your VSCODE version)

2. If the ts version is greater than 5, then you can choose to configure moduleResolution as a bundler, but in this case, other dependencies (such as Element Plus) may report crazy errors (waiting for the official fix of this problem...). ）。 You can also select "// @ts-expect-error" to ignore this error, and it will build normally after ignoring it

3. TS is less than 5, and it is the most uncomfortable when the latest version of VSCODE, and it is almost impossible to handle this error

So to sum up, if ts is less than 5, then don't care about it, the methods given by others in issues will almost always cause build errors. If ts is greater than 5, then temporarily ignore this error with "// @ts-expect-error", so that it can build normally
